### PR TITLE
[spark] Support spark to modify the default value of the column.

### LINKED ISSUE #1675: [Feature] Support spark to modify the default value of the column.
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

The spark sql statement supports modifying the default value of the column.

for example:

```sql
ALTER TABLE testAddColumn ADD COLUMN d STRING DEFAULT 0 COMMENT 'd field';

ALTER TABLE testAlterColumnDefaultValue ALTER COLUMN c SET DEFAULT 100;
```

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by zhuangchong**
### Purpose

Linked issue: close #1675 

Support spark to modify the default value of the column.

```sql
ALTER TABLE testAddColumn ADD COLUMN d STRING DEFAULT 0 COMMENT 'd field';

ALTER TABLE testAlterColumnDefaultValue ALTER COLUMN c SET DEFAULT 100;
```

### CODE REVIEW

File: docs/content/how-to/altering-tables.md [hunk 1]
```diff
@@ -413,6 +413,48 @@ Supported Type Conversions.
...
+    'fields.col_a.default-value' = '0'
+);
+```
+
+{{< /tab >}}
+
+{{< tab "Trino" >}}
+
+```sql
+ALTER TABLE my_table SET PROPERTIES fields.col_a.default-value = '0';
```
Comment (JingsongLi) [inline docs/content/how-to/altering-tables.md:449]: Have you tested this?

File: paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkCatalog.java [hunk 2]
```diff
@@ -315,48 +320,77 @@ public boolean dropTable(Identifier ident) {
...
-                    toPaimonType(add.dataType()).copy(add.isNullable()),
-                    add.comment(),
-                    move);
+            schemaChanges.add(
+                    SchemaChange.addColumn(
+                            add.fieldNames()[0],
+                            toPaimonType(add.dataType()).copy(add.isNullable()),
+                            add.comment(),
+                            move));
+            if (Objects.nonNull(add.defaultValue())) {
```
Comment (JingsongLi) [inline paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkCatalog.java:345]: `add.defaultValue()` is only working for Spark 3.4.

File: docs/content/how-to/altering-tables.md (hunk 1 above)
Comment (zhuangchong) [reply docs/content/how-to/altering-tables.md:449]: I copied this part.

https://github.com/apache/incubator-paimon/blob/cfddb74dd34fc51ff8a76b46bf54a3ed8e597d5d/docs/content/how-to/altering-tables.md?plain=1#L55-L61

Comment (zhuangchong) [reply docs/content/how-to/altering-tables.md:449]: I found that what I wrote was wrong, I will delete this part.

File: paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkCatalog.java (hunk 2 above)
Comment (zhuangchong) [reply paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkCatalog.java:345]: `UpdateColumnDefaultValue` also exists in spark 3.4, how should I deal with this situation？
is to close this PR Or is it written under the spark3.4 module?

Comment (JingsongLi) [reply paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkCatalog.java:345]: It is difficult to support both 3.3 and 3.4.
I think we can just close this PR now. Until we really need this feature.

Comment (zhuangchong) [reply paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkCatalog.java:345]: Thanks.

**Pull request closed without merging**